### PR TITLE
Backport 4.x validation

### PR DESCRIPTION
**Pull request by saeideng**
Refs  #12436
this will break your application if you are using 
https://github.com/FriendsOfCake/cakephp-upload + these methods

cc @chinpei215

### CODE REVIEW

**Review by dereuromark — APPROVED** [review]
Thank you for this

**Comment by saeideng** [comment]
this is not full BC probably :)
but fixes the incorrect use of these methods

**Comment by saeideng** [comment]
```php
//before this change
Validation::maxLength(['abc'], ....) //true 
//now
Validation::maxLength(['abc'], ....) //false

**Comment by dereuromark** [comment]
Running any of them on an array shouldn't have been done in the first place.
But if we wanted, we could also ship this with 3.8 then?

**Review by markstory — APPROVED** [review]
Looks good to me. All of these methods would emit warning or notice errors without these checks.

File: src/Validation/Validation.php [hunk 1]
```diff
@@ -876,6 +884,10 @@ public static function maxLength($check, $max)
      */
     public static function minLengthBytes($check, $min)
     {
+        if (!is_scalar($check)) {
```
Comment (inoas) [inline src/Validation/Validation.php]: Couldn't a resource be checked against its byte length?

Comment (markstory) [reply src/Validation/Validation.php]: How would a resource come in request data?

**Pull request merged**